src/webview/js/js.js

```diff
@@ -619,7 +619,16 @@ const eventUpdateHandlers = {
 const handleMessageEvent: MessageEventListener = e => {
   scrollEventsDisabled = true;
   const decodedData = decodeURIComponent(escape(window.atob(e.data)));
-  const updateEvents: WebViewUpdateEvent[] = JSON.parse(decodedData);
+  const rawUpdateEvents = JSON.parse(decodedData);
+  const updateEvents: WebViewUpdateEvent[] = rawUpdateEvents.map(updateEvent => ({
+    ...updateEvent,
+    // A URL object doesn't round-trip through JSON; we get the string
+    // representation. So, "revive" it back into a URL object.
+    ...(updateEvent.auth
+      ? { auth: { ...updateEvent.auth, realm: new URL(updateEvent.auth.realm) } }
+      : {}),
+  }));
```

Comment:
Hmm, kind of ugly, isn't it. :-/ Seems fine for a workaround, and I don't have any immediate ideas for a way to do this (and the similar spot in `handleInitialLoad`) more cleanly without a fair amount more code; but we'll want to keep an eye on this area whenever adding more complexity to it.

Reply:
Yeah, my thoughts exactly. 😦